internal/service/base/resource_custom_domain_verify.go

```diff
@@ -178,6 +179,16 @@ func (r *CustomDomainVerifyResource) Create(ctx context.Context, req resource.Cr
 
 						return diags
 					}
+
+					m, _ = regexp.MatchString("^custom domain does not need to be verified$", details[0].GetMessage())
```

Comment:
To align with the other checks, it would be best to remove the `$` from the regex here